app/assets/javascripts/libs/utils.js

```diff
@@ -129,6 +130,25 @@ export function computeArrayFromBoundingBox(bb: ?BoundingBoxType): ?Vector6 {
     : null;
 }
 
+export function aggregateBoundingBox(boundingBoxes: Array<BoundingBoxObject>): BoundingBoxType {
+  const allXCoordinates = boundingBoxes
+    .map(box => box.topLeft[0])
+    .concat(boundingBoxes.map(box => box.topLeft[0] + box.width));
+  const allYCoordinates = boundingBoxes
+    .map(box => box.topLeft[1])
+    .concat(boundingBoxes.map(box => box.topLeft[1] + box.height));
+  const allZCoordinates = boundingBoxes
+    .map(box => box.topLeft[2])
+    .concat(boundingBoxes.map(box => box.topLeft[2] + box.depth));
+  const minX = Math.min(...allXCoordinates);
+  const minY = Math.min(...allYCoordinates);
+  const minZ = Math.min(...allZCoordinates);
+  const maxX = Math.max(...allXCoordinates);
+  const maxY = Math.max(...allYCoordinates);
+  const maxZ = Math.max(...allZCoordinates);
```

Comment:
Can you refactor this to avoid repetition? E.g., lines 134 to 142 can be shortened by mapping [0, 1, 2], and for lines 143 to 148 should also be a shorter way. E.g., `const min = [0, 1, 2].map(idx => Math.min(...allCoordinates[idx]))`?